src/v2/ion/ion-type-handlers/object.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Example of the option like
+ * @param {AuthenticatorOption[]} options
+ * @param {( AuthenticatorEnrollment[] || Authenticator[] )} authenticators
+ */
+const createAuthenticatorOptions = (options = []) => {
+  return options.map(option => {
+    const value = option.value && option.value.form && option.value.form.value || [];
+
+    // Each authenticator option has list of ION field.
+    // Currently we are only support merely selecting one of options
+    // rather than pop up another page to collection extra data
+    // (in order to fill value for `mutable: true; value: null` fields).
+    // The only reason of such design is to simplify widget implementation
+    // but could subject to change in later releases.
+    // Thus only surface up fields that are `required: true; mutable: false`
+    // which implies it already has `value`.
+    const valueObject = value
+      .filter(v => v.required === true && v.mutable === false)
+      .reduce((init, v) => {
+        return Object.assign(init, { [v.name]: v.value });
+      }, {});
+
+    return {
+      label: option.label,
```

Comment:
can option.label be null ? or undefined ?

Reply:
might be but didn't see use case atm.
what else we can do if `label` is null ?

Reply:
to give some default value or empty string ?

Reply:
- empty string is same as null for courage.
- feels not necessary to have default value for now.